fiftyone/utils/open_clip.py

```diff
@@ -0,0 +1,122 @@
+"""
+CLIP model wrapper for the FiftyOne Model Zoo.
+
+| Copyright 2017-2023, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+import logging
+import os
+from packaging.version import Version
+import warnings
+
+import eta.core.web as etaw
+
+import fiftyone as fo
+import fiftyone.core.models as fom
+import fiftyone.core.utils as fou
+import fiftyone.utils.torch as fout
+import fiftyone.zoo.models as fozm
+
+fou.ensure_torch()
+import torch
+
+import open_clip
+
+logger = logging.getLogger(__name__)
+
+
+class TorchOpenClipModelConfig(fout.TorchImageModelConfig, fozm.HasZooModel):
+    """Configuration for running a :class:`TorchCLIPModel`.
+
+    See :class:`fiftyone.utils.torch.TorchImageModelConfig` for additional
+    arguments.
+
+    Args:
+        context_length: the model's context length
+        text_prompt: the text prompt to use, e.g., ``"A photo of"``
+        classes (None): a list of custom classes for zero-shot prediction
+        model_name_clip (None): the model to use
```

Comment:
Can you rename `model_name_clip` to just `model_name` for brevity?

Reply:
Right now, we are using the variable model_name in the init function. (fiftyone/zoo/models/__init__.py), we should use another name for that variable. 

```python
    def download_model_if_necessary(self):
        # pylint: disable=attribute-defined-outside-init
        if not self.model_name and not self.model_path:
            raise ConfigError(
                "Either `model_name` or `model_path` must be provided"
            )

        if self.model_path is None:
            self.model_path = download_zoo_model(self.model_name)
```

Reply:
Hi @brimoor, just let me know what could be the best name of this variable `model_name_clip` because `model_name` is already in use.

Reply:
Ah okay forgot that `model_name` was already in use. How about `clip_model_name`?

Reply:
`clip_model_name` sounds good, I'll make the changes, thank you @brimoor